### PR TITLE
add ipynb - trainer flags

### PR DESCRIPTION
**Pull request by Borda**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
adding https://github.com/PyTorchLightning/pytorch-lightning/blob/master/notebooks/05-trainer-flags-overview.ipynb

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@aribornstein shall we get this notebook without running it or just reduce to GPU usage?

File: lightning_examples/trainer-flags-gpu/flags.py [hunk 1]
```diff
@@ -0,0 +1,1600 @@
...
...U7fHzt2rHnfEwBnzJjhma35Vz/36quv9hune/R89y4G2gM4YsQIvza6F/GLL74w4/TQ8P333+/3vh4S1uVp7xCw7uFrPnzwwQdy9tlny+GHH27WS/cunnvuud7JAtnp3sfdd9/dz0vtdtttN3OY2duYXxBAAIFWBAiArcAwGgEEIicQKMR45q7n/t18883mcKse4vX90b1vOuihTd1LOGnSJHO4MycnxwSu77//3rzvCYB6eNh3CPS5Tz75pFxyySXeyQIFQM98PRPpuYd6TqEONgGw+Xz18LGe+/fhhx+aQ9+67g8//LCceeaZno/2nsfoHSFiDmmfd955flYet4qKCt9J+R0BBBAIKEAADMjCSAQQiKRAoCDmmX/37t1NwKuvr/eM8vu3rKzMhL2srCzv+OnTp8ctAOohYD3/0HfQQ8bB7AFsHgAff/xx+eMf/+g7K7nsssv8AuBDDz0k1157rd80eg6jXvixbds2v/G8QAABBIIVIAAGK8V0CCAQtoAGQL1aVvfQ+f7olazr1683F4HooeB58+aZq3vHjx8v9913nzl3Ty/0OOyww8yVwbqXa/LkyeYQqQYuT6CK5R5Az0Ug//nPf0TPUdQLNvSiEV2etva++S6vB7Jv376ih5d1fXVeGiT1te8ewN69e8uxxx5rLjjRcx/1Cma9b2Hnzp2NqQZjXSa9OviJJ54QPX+SAQEEEGhPgADYnhDvI4CAtYAGQA1AzX8efPBBM289x++mm26Sgw8+2FzIcPLJJ4teMOG5yGHixImiV73qbVfOOOMMmTp1qplXPAKgLnDz28D079/fLE9NTU2rVoECYG1trQm6eohZ1/2RRx6R1NRUvwCoN6rWq6L1SmSdh+c2MHpF8D333GPOHVSX448/XnRvIXsFW90EvIEAAj4CBEAfDH5FAAEEwhF4+eWXzV7AcNrSBgEEEIiHAAEwHup8JgIIJLSAXrShh6v1fn6DBw82N2PWW8EwIIAAAokiQABMlC3FciKAgGME9PD0kUceaQ5J67l4vXr1ktYuYnHMQrMgCCCAgI8AAdAHg18RQAABBBBAAAE3CBAA3bCVWUcEEEAAAQQQQMBHgADog8GvCCCAAAIIIICAGwQIgG7YyqwjAggggAACCCDgI0AA9MHgVwQQQAABBBBAwA0C/w+ELQeExqjNywAAAABJRU5ErkJggg==)
```
Comment (Borda) [inline lightning_examples/trainer-flags-gpu/flags.py:1026]: make the image a stand-alone image in the repo/folder

File: lightning_examples/trainer-flags-gpu/flags.py [hunk 2]
```diff
@@ -0,0 +1,1600 @@
...
+    def test_dataloader(self):
+        return DataLoader(self.mnist_test, batch_size=self.batch_size, num_workers=self.num_workers)
+
+
+mnist = MNISTDataModule()
+
+# %% [markdown] id="gEulmrbxwaYL"
+# ### Simple AutoEncoder Model
+#
+# Were gonna define a simple Lightning model so we can play with all the settings of the Lightning Trainer.
```
Comment (ethanwharris) [inline lightning_examples/trainer-flags-gpu/flags.py:81]:
```suggestion
# We're gonna define a simple Lightning model so we can play with all the settings of the Lightning Trainer.
```

File: lightning_examples/trainer-flags-gpu/flags.py [hunk 3]
```diff
@@ -0,0 +1,1600 @@
...
+#
+# LightningModule is simply pure Pytorch reorganized into hooks, that represents all the steps in the training process.
```
Comment (ethanwharris) [inline lightning_examples/trainer-flags-gpu/flags.py:83]:
```suggestion
# LightningModule is simply pure Pytorch reorganized into hooks that represent all the steps in the training process.
```

File: lightning_examples/trainer-flags-gpu/flags.py [hunk 4]
```diff
@@ -0,0 +1,1600 @@
...
+#
+# You can use LightningModule hooks to control every part of your model, but for the purpose of this video
```
Comment (ethanwharris) [inline lightning_examples/trainer-flags-gpu/flags.py:85]:
```suggestion
# You can use LightningModule hooks to control every part of your model. For the purpose of this tutorial
```

File: lightning_examples/trainer-flags-gpu/flags.py [hunk 5]
```diff
@@ -0,0 +1,1600 @@
...
+# we will use a very simple MNIST classifier, a model that takes 28*28 grayscale images of hand written images,
```
Comment (ethanwharris) [inline lightning_examples/trainer-flags-gpu/flags.py:86]:
```suggestion
# we will use a very simple MNIST classifier. Our model takes 28*28 grayscale images of hand written images
```

File: lightning_examples/trainer-flags-gpu/.meta.yml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+title: Introduction to Lightning Flags
+author: PL team
+created: 2021-01-31
+updated: 2021-02-23
+license: CC
+description: >
+  In this notebook, we'll go over the flags available in the `Trainer` object.
+  Note that not everything will work in the Colab environment (multi-gpu, etc).
+  This notebook accompanies the Trainer videos we'll be putting out.
```
Comment (ethanwharris) [inline lightning_examples/trainer-flags-gpu/.meta.yml:9]: Are the videos out now?
```suggestion
  This notebook accompanies our Trainer videos.
```

**Review by ethanwharris — COMMENTED** [review]
Was this notebook just transcribed directly from the video? There are a lot of spurious commas and very long sentences. Also, it's a bit weird that the text says "in this video" a lot when it's a notebook / tutorial haha. @Borda Are these grammar corrections and stuff in scope for this PR?

File: lightning_examples/trainer-flags-gpu/flags.py [hunk 7]
```diff
@@ -0,0 +1,1600 @@
...
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py:percent
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.11.2
+# ---
+
```
Comment (Borda) [inline lightning_examples/trainer-flags-gpu/flags.py:12]:
```suggestion
```

**Comment by Borda** [comment]
> Was this notebook just transcribed directly from the video? There are a lot of spurious commas and very long sentences. Also, it's a bit weird that the text says "in this video" a lot when it's a notebook / tutorial haha. @Borda Are these grammar corrections and stuff in scope for this PR?

not sure, I took it from our PL repo, maybe @edenlightning could bring some light?

**Pull request closed without merging**